packages/shared_preferences/lib/shared_preferences_tizen.dart

```diff
@@ -101,39 +99,56 @@ class SharedPreferencesPlugin extends SharedPreferencesStorePlatform {
 
   @override
   Future<bool> remove(String key) async {
-    _preferences.remove(key);
-
-    return bindings.remove(key.toNativeUtf8()) == 0;
+    return using((Arena arena) {
+      final bool ret = bindings.remove(key.toNativeUtf8(allocator: arena)) == 0;
+      if (ret) {
+        _preferences.remove(key);
+      }
+      return ret;
+    });
   }
 
   @override
   Future<bool> setValue(String valueType, String key, Object value) async {
     _preferences[key] = value;
 
-    final Pointer<Utf8> pKey = key.toNativeUtf8();
-    int ret;
-    if (valueType == 'Bool') {
-      ret = bindings.setBoolean(pKey, (value as bool) ? 1 : 0);
-    } else if (valueType == 'Double') {
-      ret = bindings.setDouble(pKey, value as double);
-    } else if (valueType == 'Int') {
-      ret = bindings.setInt(pKey, value as int);
-    } else if (valueType == 'String') {
-      ret = bindings.setString(pKey, (value as String).toNativeUtf8());
-    } else if (valueType == 'StringList') {
-      // Tizen Preference API doesn't support arrays.
-      final List<String> list = value as List<String>;
-      String joined;
-      if (list.isEmpty) {
-        joined = _separator;
-      } else {
-        joined = _separator + list.join(_separator) + _separator;
+    return using((Arena arena) {
+      int ret;
+      final Pointer<Utf8> pKey = key.toNativeUtf8(allocator: arena);
+      switch (valueType) {
+        case 'Bool':
+          ret = bindings.setBoolean(pKey, (value as bool) ? 1 : 0);
+          break;
```

Comment:
The more code fitting in one screen, the better readability.
```suggestion
        case 'Bool':
          return bindings.setBoolean(pKey, (value as bool) ? 1 : 0);
```
(`== 0` should be added at line 146.)